src/Raven.Server/Documents/PeriodicBackup/Aws/RavenAwsClient.cs

```diff
@@ -25,21 +22,38 @@ public abstract class RavenAwsClient : RavenStorageClient
 
         public const string DefaultRegion = "us-east-1";
 
-        private static bool _endpointsLoaded;
-
-        private static readonly Dictionary<string, string> AwsRegionNames = new Dictionary<string, string>();
+        private static readonly HashSet<string> AwsRegionNames = new HashSet<string>
+        {
+            "us-east-1",
+            "us-east-2",
+            "us-west-1",
+            "us-west-2",
+            "ca-central-1",
+            "eu-west-1",
+            "eu-central-1",
+            "ap-northeast-1",
+            "ap-northeast-2",
+            "ap-southeast-1",
+            "ap-southeast-2",
+            "ap-south-1",
+            "sa-east-1",
+            "us-gov-west-1"
+        };
 
         private readonly string _awsAccessKey;
         private readonly byte[] _awsSecretKey;
-        private readonly string _awsRegionName;
 
-        protected string AwsRegion { get; private set; }
+        protected string AwsRegion { get; }
 
         protected RavenAwsClient(string awsAccessKey, string awsSecretKey, string awsRegionName)
         {
+            awsRegionName = awsRegionName.ToLower();
+            if (AwsRegionNames.Contains(awsRegionName) == false)
+                throw new InvalidOperationException("Given endpoint is invalid: " + awsRegionName);
```

Comment:
Not a good idea, this requires us to deploy new version whenever amazon have a new data center